pkg/kclient/ingress_test.go

```diff
@@ -0,0 +1,236 @@
+package kclient
+
+import (
+	"testing"
+
+	"github.com/pkg/errors"
+	extensionsv1 "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	ktesting "k8s.io/client-go/testing"
+)
+
+func TestCreateIngress(t *testing.T) {
+
+	tests := []struct {
+		name        string
+		ingressName string
+		wantErr     bool
+	}{
+		{
+			name:        "Case: Valid ingress name",
+			ingressName: "testIngress",
+			wantErr:     false,
+		},
+		{
+			name:        "Case: Invalid ingress name",
+			ingressName: "",
+			wantErr:     true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// initialising the fakeclient
+			fkclient, fkclientset := FakeNew()
+			fkclient.Namespace = "default"
+
+			objectMeta := CreateObjectMeta(tt.ingressName, "default", nil, nil)
+
+			fkclientset.Kubernetes.PrependReactor("create", "ingresses", func(action ktesting.Action) (bool, runtime.Object, error) {
```

Comment:
we don't need a prependReactor in case of create. To replicate the scenario of empty ingress name, check and throw the error from the create function itself. That way we can save a API call.

Reply:
updated